spring-integration-core/src/main/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiator.java

```diff
@@ -111,6 +112,13 @@
 	 */
 	private long busyWaitMillis = DEFAULT_BUSY_WAIT_TIME;
 
+	/**
+	 * A boolean that if true, failures that occur during leader election will
+	 * be published to the specified applicationEventPublisher.  If false,
+	 * no failures will be published.  Default is false.
```

Comment:
Sounds like should be moved to the setters JavaDoc.
Plus `@since 5.0`

Also need to give people a hint "why?".
Really: why can't we emit such an event unconditionally all the time we fail to obtain the lock?

Thanks

Reply:
Will move the comment to the setPublishFailedEvents.
The flag was Gary's idea, so we won't flood the users with these events when its in the loop.